### PR TITLE
feat(spv-503) syning spv-wallet values for helm

### PR DESCRIPTION
**Pull request by ac4ch**

### CODE REVIEW

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 1]
```diff
@@ -123,22 +124,29 @@ config:
...
   # enable (ITC) incoming transaction checking
   disable_itc: true
-  nodes:
-    # deployment id used annotating api calls in XDeployment-ID header - this value will be randomly generated if not set
-    deployment_id: ''
+  
+  arc:
+    url: https://arc.taal.com
+    token: mainnet_06770f425eb00298839a24a49cbdc02c
+    _deployment_id: spv-wallet-deployment-id
```
Comment (chris-4chain) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: I'm just wondering if we need to hold those "commented out" (underscored) params.
I'm not very familiar with this repository - so, just a thought.

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 2]
```diff
@@ -8,91 +8,100 @@ data:
   config.yaml: |-
     auth:
       require_signing: {{ .Values.config.auth.require_signing }}
-      scheme: {{ .Values.config.auth.scheme }} 
+      scheme: {{ .Values.config.auth.scheme }}
       signing_disabled: {{ .Values.config.auth.signing_disabled }}
-      admin_key: ""
+      admin_key: {{ .Values.config.auth.admin_key }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: we're setting this value as environment variable so, we don't need it here.
Please revert this change and add a comment that we're setting it as environment variable.

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 3]
```diff
@@ -8,91 +8,100 @@ data:
...
+        user: {{ .Values.postgresql.auth.username }}
+      sqlite:
+        database_path: {{ .Values.config.db.sqlite.database_path }}
+        debug: {{ .Values.config.db.sqlite.debug }}
+        max_connection_idle_time: {{ .Values.config.db.sqlite.max_connection_idle_time }}
+        max_connection_time: {{ .Values.config.db.sqlite.max_connection_time }}
+        max_idle_connections: {{ .Values.config.db.sqlite.max_idle_connections }}
+        max_open_connections: {{ .Values.config.db.sqlite.max_open_connections }}
+        shared: {{ .Values.config.db.sqlite.shared }}
+    debug: {{ .Values.config.debug }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: debug was remove recently

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 4]
```diff
@@ -8,91 +8,100 @@ data:
...
+    debug_profiling: {{ .Values.config.debug_profiling }}
+    disable_itc: {{ .Values.config.disable_itc }}
+    arc:
+      url: {{ .Values.config.arc.url }}
+      token: {{ .Values.config.arc.token }}
+      _deployment_id: {{ .Values.config.arc._deployment_id }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: it was generated previously (look at lines 54-58 on file before changes)

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 5]
```diff
@@ -8,91 +8,100 @@ data:
...
       callback:
-        callback_host: 'https://{{.Values.global.domainName}}'
-      use_fee_quotes: {{ .Values.config.nodes.use_fee_quotes }}
+        enabled: {{ .Values.config.arc.callback.enabled }}
+        host: {{ .Values.config.arc.callback.host }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: the host was automatically resolved from domainName

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 6]
```diff
@@ -8,91 +8,100 @@ data:
...
+        _token: {{ .Values.config.arc.callback._token }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: it should be rather optionally added, something like in pseudocode below:
```
if (.Values.config.arc.callback.token) {
  token: .Values.config.arc.callback._token
}
```

Bear in mind that the `_` shouldn't be used, because we use it in example config to show example but not use that static value, so the valid key is really without `_`

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 7]
```diff
@@ -8,91 +8,100 @@ data:
...
+      use_fee_quotes: {{ .Values.config.arc.use_fee_quotes }}
       fee_unit:
-        satoshis: {{ .Values.config.nodes.fee_unit.satoshis }}
-        bytes: {{ .Values.config.nodes.fee_unit.bytes }}
+        satoshis: {{ .Values.config.arc.fee_unit.satoshis }}
+        bytes: {{ .Values.config.arc.fee_unit.bytes }}
     notifications:
       enabled: {{ .Values.config.notifications.enabled }}
       webhook_endpoint: "{{ .Values.config.notifications.webhook_endpoint }}"
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: There is no webhook_endpoint property anymore

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 8]
```diff
@@ -8,91 +8,100 @@ data:
...
+    block_headers_service:
+      auth_token: {{ .Values.config.block_headers_service.auth_token }}
+      url: {{ .Values.config.block_headers_service.url }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: we should calculate `block_headers_service_url`. If the `.Values.config.block_headers_service.url` is set, then use that value, if not, then calculate the value as `http://{{- printf "%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" }}`

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 9]
```diff
@@ -8,91 +8,100 @@ data:
...
     paymail:
       beef:
-        {{- if .Values.config.paymail.beef.block_headers_service_url }}
+        block_headers_service_auth_token: {{ .Values.config.paymail.beef.block_headers_service_auth_token }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: this is set as environment variable

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 10]
```diff
@@ -8,91 +8,100 @@ data:
...
       fee_unit:
-        satoshis: {{ .Values.config.nodes.fee_unit.satoshis }}
-        bytes: {{ .Values.config.nodes.fee_unit.bytes }}
+        satoshis: {{ .Values.config.arc.fee_unit.satoshis }}
+        bytes: {{ .Values.config.arc.fee_unit.bytes }}
     notifications:
       enabled: {{ .Values.config.notifications.enabled }}
       webhook_endpoint: "{{ .Values.config.notifications.webhook_endpoint }}"
+    block_headers_service:
+      auth_token: {{ .Values.config.block_headers_service.auth_token }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: similar to old setup for block headers service, this should be set as environment variable, which is taking a value from secret.

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 11]
```diff
@@ -8,91 +8,100 @@ data:
...
+      url: {{ .Values.config.block_headers_service.url }}
     paymail:
       beef:
-        {{- if .Values.config.paymail.beef.block_headers_service_url }}
+        block_headers_service_auth_token: {{ .Values.config.paymail.beef.block_headers_service_auth_token }}
         block_headers_service_url: {{ .Values.config.paymail.beef.block_headers_service_url }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: we should calculate block_headers_service_url.
 If the `.Values.config.block_headers_service.url` is set, then use that value + `/api/v1/chain/merkleroot/verify` suffix, 
if not, then calculate the value as `'http://{{- printf "%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" }}/api/v1/chain/merkleroot/verify'`

Notice that this value is sligthly different because it needs to have also this suffix `/api/v1/chain/merkleroot/verify`

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 12]
```diff
@@ -8,91 +8,100 @@ data:
...
-        {{- else }}
-        block_headers_service_url: 'http://{{- printf "%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" }}/api/v1/chain/merkleroot/verify'
-        {{- end }}
-        use_beef: {{  .Values.config.paymail.beef.use_beef }}
-      default_from_paymail: default@{{  .Values.global.domainName }}
-      default_note: {{  .Values.config.paymail.default_note }}
-      domain_validation_enabled: {{  .Values.config.paymail.domain_validation_enabled }}
+        use_beef: {{ .Values.config.paymail.beef.use_beef }}
+      default_from_paymail: {{ .Values.config.paymail.default_from_paymail }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: it was calculated

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml (hunk 1 above)
Comment (dorzepowski) [reply charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: no we don't need it especially as previously there was such an entry and we should keep this solution:
```yaml
   # deployment id used annotating api calls in XDeployment-ID header - this value will be randomly generated if not set
    deployment_id: ''
```

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 13]
```diff
@@ -123,22 +124,29 @@ config:
...
+    url: https://arc.taal.com
+    token: mainnet_06770f425eb00298839a24a49cbdc02c
+    _deployment_id: spv-wallet-deployment-id 
     callback:
-      # set token to authenticate callback calls, by default callback token will be generated
-      callback_token: ''
-    # use fee quotes for transaction fee calculation
-    use_fee_quotes: true
-    # used as the fee value if 'use_fee_quotes' is set to false
+      enabled: false
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: it should be enabled by default in helm chart

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 14]
```diff
@@ -123,22 +124,29 @@ config:
...
+    token: mainnet_06770f425eb00298839a24a49cbdc02c
+    _deployment_id: spv-wallet-deployment-id 
     callback:
-      # set token to authenticate callback calls, by default callback token will be generated
-      callback_token: ''
-    # use fee quotes for transaction fee calculation
-    use_fee_quotes: true
-    # used as the fee value if 'use_fee_quotes' is set to false
+      enabled: false 
+      host: ''
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: it's not needed because it should be calculated from domain, as previously

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 15]
```diff
@@ -123,22 +124,29 @@ config:
...
     callback:
-      # set token to authenticate callback calls, by default callback token will be generated
-      callback_token: ''
-    # use fee quotes for transaction fee calculation
-    use_fee_quotes: true
-    # used as the fee value if 'use_fee_quotes' is set to false
+      enabled: false 
+      host: '' 
+      _token: ''  
+    use_fee_quotes: true
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: revert those 2 to previous keys/values with comments

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 16]
```diff
@@ -123,22 +124,29 @@ config:
...
-    # use fee quotes for transaction fee calculation
-    use_fee_quotes: true
-    # used as the fee value if 'use_fee_quotes' is set to false
+      enabled: false 
+      host: '' 
+      _token: ''  
+    use_fee_quotes: true  
     fee_unit:
-      satoshis: 1
+      satoshis: 1
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: not needed space at the end of the line

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 17]
```diff
@@ -123,22 +124,29 @@ config:
...
       bytes: 1000
+  
   notifications:
     enabled: false
     # url to send notifications
     webhook_endpoint: ''
+  
+  block_headers_service:
+    auth_token: ''
+    url: http://localhost:8080
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: we should use empty string and put a comment that by default it will try to connect to BHS in cluster.

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 18]
```diff
@@ -148,6 +156,7 @@ config:
     default_note: SPV Wallet Address Resolution
     # enable paymail domain validation, paymail domain must be in domains list to be valid and that the transaction can be processed
     domain_validation_enabled: true
+    default_from_paymail: ''
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: hopefully we will remove it soon, so let's keep it as it was (calculated without option to set it in values).

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 19]
```diff
@@ -111,6 +111,7 @@ config:
       max_open_connections: 0
       # it must be set here at least as empty otherwise we won't be able to override it with env
       password: ""
+      user: postgres
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: in configmap it is taken from the postgres chart dependency so it's even not used.

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 20]
```diff
@@ -111,6 +111,7 @@ config:
       max_open_connections: 0
       # it must be set here at least as empty otherwise we won't be able to override it with env
       password: ""
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: I'm not sure if we use it anywhere.

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 21]
```diff
@@ -8,91 +8,110 @@ data:
...
+        deployment_id: 'spv-wallet-{{ printf "%s-$s"  (include "spv-wallet.fullname" .) .Values.global.domainName | adler32sum }}'
+      {{- end }}
       callback:
-        callback_host: 'https://{{.Values.global.domainName}}'
-      use_fee_quotes: {{ .Values.config.nodes.use_fee_quotes }}
+        enabled: {{ .Values.config.arc.callback.enabled }}
+        host: 'https://{{.Values.global.domainName}}'
+        if (.Values.config.arc.callback.token) {  
+          token: .Values.config.arc.callback.token  
+        }
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: it was only pseudo code 😄 
you need to use go template if version

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 22]
```diff
@@ -8,91 +8,110 @@ data:
...
-        {{- if .Values.config.paymail.beef.block_headers_service_url }}
-        block_headers_service_url: {{ .Values.config.paymail.beef.block_headers_service_url }}
-        {{- else }}
-        block_headers_service_url: 'http://{{- printf "%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" }}/api/v1/chain/merkleroot/verify'
-        {{- end }}
-        use_beef: {{  .Values.config.paymail.beef.use_beef }}
+        block_headers_service_auth_token: ""
+        {{- $defaultUrl := printf "http://%s-block-headers-service/api/v1/chain/merkleroot/verify" .Release.Name | trunc 63 | trimSuffix "-" -}}
+        {{- $blockHeadersServiceUrl := printf "%s/api/v1/chain/merkleroot/verify" .Values.config.block_headers_service.url | default $defaultUrl -}}
+        url: {{ $blockHeadersServiceUrl }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: Can't we simply reuse already calculated `$blockHeadersServiceUrl` and add a static suffix, something like:
"{{ $blockHeadersServiceUrl }}/api/v1/chain/merkleroot/verify"

Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: I think that the property name should be `block_headers_service_url` not `url` in line 91

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 23]
```diff
@@ -123,22 +123,34 @@ config:
...
+    token: mainnet_06770f425eb00298839a24a49cbdc02c
+    # deployment id used annotating api calls in XDeployment-ID header - this value will be randomly generated if not set  
+    deployment_id: ''  
+ 
+    # set token to authenticate callback calls, by default callback token will be generated randomly
     callback:
-      # set token to authenticate callback calls, by default callback token will be generated
-      callback_token: ''
+      enabled: true 
+      token: ''
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: move the comment about token to be just above token key

File: charts/spv-wallet-stack/charts/spv-wallet/values.yaml [hunk 24]
```diff
@@ -123,22 +123,34 @@ config:
...
     use_fee_quotes: true
     # used as the fee value if 'use_fee_quotes' is set to false
     fee_unit:
       satoshis: 1
       bytes: 1000
+  
   notifications:
     enabled: false
     # url to send notifications
     webhook_endpoint: ''
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/values.yaml]: this can be removed, because it's not used anymore

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml (hunk 21 above)
Comment (nonus25) [reply charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: :)

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml (hunk 22 above)
Comment (nonus25) [reply charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: yes it should be 
```yaml
block_headers_service:
  auth_token: mQZQ6WmxURxWz5ch
  # URL used to communicate with Block Headers Service (BHS)
  url: http://localhost:8080
paymail:
  beef:
    block_headers_service_auth_token: mQZQ6WmxURxWz5ch
    # url to Block Headers Service, used for merkle root verification
    block_headers_service_url: http://localhost:8080/api/v1/chain/merkleroot/verify
    use_beef: false
```

Comment (ac4ch) [reply charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: did this way, hope this wold do it `{{- $beefBlockHeadersServiceUrl := printf "%s/api/v1/chain/merkleroot/verify" $blockHeadersServiceUrl -}}`

**Comment by dorzepowski** [comment]
Testing

**Comment by dorzepowski** [comment]
There are still refereneces to `.Values.config.nodes` which was replaced with `arc` and this makes the helm chart templating failing with error.
I'm getting failures on: 
configmap.yml line 63 and 64, 
secret.yml line 10 and 11

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 25]
```diff
@@ -8,91 +8,109 @@ data:
...
+        password: {{ .Values.config.db.sql.password }}
+        port: {{ .Values.config.db.sql.port }}
+        replica: {{ .Values.config.db.sql.replica }}
+        skip_initialize_with_version: {{ .Values.config.db.sql.skip_initialize_with_version }}
+        table_prefix: {{ .Values.config.db.sql.table_prefix }}
+        time_zone: {{ .Values.config.db.sql.time_zone }}
+        tx_timeout: {{ .Values.config.db.sql.tx_timeout }}
+        user: {{ .Values.postgresql.auth.username }}
+      sqlite:
+        database_path: {{ .Values.config.db.sqlite.database_path }}
```
Comment (jakubmkowalski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml:53]: `nil pointer evaluating interface {}.database_path`
Shouldn't we have a configuration for sqlite in values.yaml?

Comment (nonus25) [reply charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml:53]: i know i have fix for this

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 26]
```diff
@@ -8,91 +8,114 @@ data:
...
+        debug: {{ .Values.config.db.sqlite.debug }}
+        max_connection_idle_time: {{ .Values.config.db.sqlite.max_connection_idle_time }}
+        max_connection_time: {{ .Values.config.db.sqlite.max_connection_time }}
+        max_idle_connections: {{ .Values.config.db.sqlite.max_idle_connections }}
+        max_open_connections: {{ .Values.config.db.sqlite.max_open_connections }}
+        shared: {{ .Values.config.db.sqlite.shared }}
+      {{- else }}
+      error: "Unsupported database engine specified. Please use 'postgresql' or 'sqlite'."
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: Use fail function instead https://helm.sh/docs/chart_template_guide/function_list/#fail

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 27]
```diff
@@ -8,91 +8,114 @@ data:
...
+    debug_profiling: {{ .Values.config.debug_profiling }}
+    disable_itc: {{ .Values.config.disable_itc }}
+    arc:
+      url: {{ .Values.config.arc.url }}
+      token: {{ .Values.config.arc.token }}
+      {{- if .Values.config.arc.deployment_id }}
+        deployment_id: {{ .Values.config.arc.deployment_id }}
+      {{- else }}
+        deployment_id: 'spv-wallet-{{ printf "%s-$s"  (include "spv-wallet.fullname" .) .Values.global.domainName | adler32sum }}'
+      {{- end }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: deployment_id has invalid indentation

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 28]
```diff
@@ -8,91 +8,114 @@ data:
...
+        satoshis: {{ .Values.config.arc.fee_unit.satoshis }}
+        bytes: {{ .Values.config.arc.fee_unit.bytes }}
     notifications:
       enabled: {{ .Values.config.notifications.enabled }}
-      webhook_endpoint: "{{ .Values.config.notifications.webhook_endpoint }}"
+    block_headers_service:
+      auth_token: ""
+      {{- $defaultUrl := printf "http://%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" -}}
+      {{- $blockHeadersServiceUrl := default $defaultUrl .Values.config.block_headers_service.url -}}
+      url: {{ $blockHeadersServiceUrl }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: url property is generated in the same line as auth_token (probably because you don't need to use `-}}` but simply `}}` )

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 29]
```diff
@@ -8,91 +8,114 @@ data:
...
       beef:
-        {{- if .Values.config.paymail.beef.block_headers_service_url }}
-        block_headers_service_url: {{ .Values.config.paymail.beef.block_headers_service_url }}
-        {{- else }}
-        block_headers_service_url: 'http://{{- printf "%s-block-headers-service" .Release.Name | trunc 63 | trimSuffix "-" }}/api/v1/chain/merkleroot/verify'
-        {{- end }}
-        use_beef: {{  .Values.config.paymail.beef.use_beef }}
+        block_headers_service_auth_token: ""
+        {{- $beefBlockHeadersServiceUrl := printf "%s/api/v1/chain/merkleroot/verify" $blockHeadersServiceUrl -}}
+        block_headers_service_url: {{ $beefBlockHeadersServiceUrl }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml:95]: similiar to url above, block_headers_service_url is generated in the same line as block_headers_service_auth_token

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 30]
```diff
@@ -8,91 +8,114 @@ data:
...
+        use_tls: {{ .Values.config.cache.redis.use_tls }}
     db:
       datastore:
-        debug: {{  .Values.config.db.datastore.debug }} 
-        engine: {{  .Values.config.db.datastore.engine }} 
-        table_prefix: {{  .Values.config.db.datastore.table_prefix }} 
+        debug: {{ .Values.config.db.datastore.debug }}
+        engine: {{ .Values.config.db.datastore.engine }}
+        table_prefix: {{ .Values.config.db.datastore.table_prefix }}
+      {{- if eq .Values.config.db.datastore.engine "postgres" }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: the value for engine must be postgresql - currently it always end up in the else statement for postgresql configuration.

```suggestion
      {{- if eq .Values.config.db.datastore.engine "postgresql" }}
```

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 31]
```diff
@@ -58,17 +58,17 @@ data:
         max_open_connections: {{ .Values.config.db.sqlite.max_open_connections }}
         shared: {{ .Values.config.db.sqlite.shared }}
       {{- else }}
-      error: "Unsupported database engine specified. Please use 'postgresql' or 'sqlite'."
+      fail: "Unsupported database engine specified. Please use 'postgresql' or 'sqlite'."
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]:
```suggestion
      {{ fail "Unsupported database engine specified. Please use 'postgresql' or 'sqlite'." }}
```

File: charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml [hunk 32]
```diff
@@ -8,91 +8,114 @@ data:
...
       deployment_id: 'spv-wallet-{{ printf "%s-$s"  (include "spv-wallet.fullname" .) .Values.global.domainName | adler32sum }}'
-    {{- end }}
+      {{- end }}
       callback:
-        callback_host: 'https://{{.Values.global.domainName}}'
-      use_fee_quotes: {{ .Values.config.nodes.use_fee_quotes }}
+        enabled: {{ .Values.config.arc.callback.enabled }}
+        host: 'https://{{.Values.global.domainName}}'
+        {{- if .Values.config.arc.callback.token }}
+          token: {{ .Values.config.arc.callback.token }}
```
Comment (dorzepowski) [inline charts/spv-wallet-stack/charts/spv-wallet/templates/configmap.yml]: this also will have invalid indentation

**Review by dorzepowski — CHANGES_REQUESTED** [review]

**Review by dorzepowski — APPROVED** [review]
